src/main/java/com/teragrep/pth10/steps/spath/SpathEscapedKey.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Teragrep Data Processing Language (DPL) translator for Apache Spark (pth_10)
+ * Copyright (C) 2019-2024 Suomen Kanuuna Oy
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.pth10.steps.spath;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+public final class SpathEscapedKey implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+    private final String key;
+
+    public SpathEscapedKey(final String key) {
+        this.key = key;
+    }
+
+    public SpathKey unescaped() {
```

Comment:
I was thinking this during the day ans thought that is this actually an escaped thing or is it rather a backtick quoted string and usual unquote functionality should.be used? @eemhu Any comments as you did previous quotation fixes

Reply:
I suppose strictly speaking it is more of a backtick quotation but logically it is used to escape `.` (dots), so they can be used in the column names.

Reply:
I think we should make the object re-usable for such purposes and use the generic quotation object that allows specifying the quotation character. this way it would be able to handle 
```
' " `
```
 and possibly other characters as well. current substring way does not handle properly escapes in a string. i.e. 
```
`foo\`bar`
```
 should be unquoted as 
```
foo`bar
```

Reply:
Added backtick support for existing UnquotedText object and also added QuotedText object.